content/cloud-hosting/azure/azure-service-bus.md

```diff
@@ -0,0 +1,254 @@
+var username = WebUtility.UrlEncode("");   // Azure Service Bus is PolicyName
+var password = WebUtility.UrlEncode("");   // Azure Service Bus is Primary Key or secondary key
+// Connection string to connect Azure Service Bus or RabmitMQ
+var connectionString = $"{protocal}://{username}:{password}@{hostname}/";
+// Make connection
+var connection = new Connection(new Address(connectionString));
+// Make session
+var amqpSession = new Session(connection);
+// set sender id
+var senderSubscriptionId = "thedemo.amqp.sender";
+// this is queue name
+var queueName = "poc01";
+// Make Link
+var sender = new SenderLink(amqpSession, senderSubscriptionId, queueName);
+for (var i = 0; i < 10; i++)
+{
+// Create new message
+var message = new Message($"Hello Message Queue {i}");
+// Set message id
+message.Properties = new Properties() { MessageId = Guid.NewGuid().ToString() };
+// Add custom property to message
+message.ApplicationProperties = new ApplicationProperties();
+message.ApplicationProperties["Message.Type.Method"] = "Demo Property";
+// Send message to service bus
+sender.Send(message);
+}
+ ```
+
+Thereciever ใช้สำหรับรับ message จาก Message Queue มีดังนี้
+
+ ```cs 
+
+using Amqp;
+using Amqp.Framing;
+using System.Xml;
+using System.Net;
+var protocal = "amqp";              // use amqps for Azure Service Bus, use amqp for RabmitMQ in local
+var hostname = "localhost:5672";    // use localhost:5672 for RabmitMQ in local
+var username = WebUtility.UrlEncode("");   // Azure Service Bus is PolicyName
+var password = WebUtility.UrlEncode("");   // Azure Service Bus is Primary Key or secondary key
+// Connection string to connect Azure Service Bus or RabmitMQ
+var connectionString = $"{protocal}://{username}:{password}@{hostname}/";
+// Make connection
+var connection = new Connection(new Address(connectionString));
+// Make session
+var amqpSession = new Session(connection);
+// set receiver id
+var receiverSubscriptionId = "thedemo.amqp.receiver";
+// this is queue name
+var queueName = "poc01";
+// Make link
+var consumer = new ReceiverLink(amqpSession, receiverSubscriptionId, $"{queueName}");
+// Register handler to manage menage when get any new message
+consumer.Start(5, OnMessageCallback);
+// Wait until press some key
+Console.Read();
+// Handler for manage activity when getting the new message
+static void OnMessageCallback(IReceiverLink receiver, Amqp.Message message)
+{
+try
+{
+// Get custom property from message
+var messageType = message.ApplicationProperties["Message.Type.Method"];
+// Read message body
+var rawBody = message.Body;
+Console.WriteLine(rawBody.ToString() + " " + messageType);
+// Accept message once compled reading
+receiver.Accept(message);
+}
+catch (Exception ex)
+{
+// Reject message in case got any error
+receiver.Reject(message);
+Console.WriteLine(ex);
+}
+}
+
+  ```
+
+ในการทดสอบกับ RabbitMQ ให้กำหนดค่า ดังนี้
+
+![](https://miro.medium.com/max/1400/1*tNtjbMmpOqc-Dln7xMFLGg.png)
```

Comment:
Could you please pull all images in this document to this repository? It is safer for backup our image files?

Here is my suggested path:
Put images in this folder
```
content/cloud-hosting/azure/images
```

Use it as
```
![](images/azure-01.png)
```

This is because image is relative to a Markdown file, so you don't need to use image full path.